src/components/top-bar/styles.ts

```diff
@@ -0,0 +1,87 @@
+import { css } from '@linaria/core'
+import { styled } from '@linaria/react'
+import { isDesktopOrBelow, isMobile, isTabletOrBelow } from '../../styles/media'
+import { ElButtonGroup } from '../button-group'
+import { ElNavSearchButton } from '../nav-search-button/styles'
+import { ElButtonNavIconItem } from '../nav-icon-item'
+
+export const elTopBarAvatar = css`
+  display: flex;
+  padding: var(--spacing-1, 4px) var(--spacing-none, 0px);
+  padding-left: var(--spacing-2, 8px);
+  align-items: center;
+  gap: 10px;
+`
+
+export const ElTopBarMainNav = styled(ElButtonGroup)`
+  display: flex;
+  padding: var(--spacing-1, 4px) var(--spacing-none, 0px) var(--spacing-1, 4px) var(--spacing-6, 24px);
+  align-items: flex-start;
+  flex-grow: 1;
+`
+
+export const ElTopBarSecondaryNav = styled(ElButtonGroup)`
+  flex-wrap: nowrap;
+`
+
+export const ElTopBar = styled.div`
+  display: flex;
+  height: var(--size-10, 56px);
+  padding: var(--spacing-2, 8px) var(--spacing-5, 20px);
+  justify-content: space-between;
+  align-items: center;
+  flex-shrink: 0;
+  border-bottom: var(--border-default, 1px) solid var(--outline-default, #e5e9ed);
+  background: var(--fill-white, #fff);
+
+  ${isDesktopOrBelow} {
+    padding: var(--spacing-2, 8px) var(--spacing-4, 16px);
+
+    // TODO: need to be adjusted once AppSwitcher and Brand/Logo included
+    justify-content: flex-end;
+
+    ${ElTopBarMainNav}, ${ElTopBarSecondaryNav} {
+      display: none;
+    }
+  }
+
+  ${isTabletOrBelow} {
+    ${elTopBarAvatar} {
+      display: none;
+    }
+  }
+`
+
+export const ElTopBarSearchContainer = styled.div`
+  padding: var(--spacing-1, 4px) var(--spacing-4, 16px) var(--spacing-1, 4px) var(--spacing-none, 0px);
+  ${ElNavSearchButton} {
```

Comment:
**issue:** Doing this couples us tightly to the `ElNavSearchButton`. We need to avoid doing this wherever possible. We don't need to set the width of `ElNavSearchButton`, we only need to set the width of `ElTopBarSearchContainer` and the component placed inside of it should respect that width.

Reply:
because the `search-button` itself has defined padding (400px), but as per previous comment I think we can either delete that defined width or add additional `max-width: 100%` to `search-button` so it can respect the container 👍

Reply:
I don't think the search button should have a fixed width. Looking at the  design spec for it, I don't think that 400px is intentional as the use of the component in the top bar maxes out at 200px.

Reply:
ok, I will delete it then 👍